src/App.js

```diff
@@ -14,11 +15,85 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    goods: [...goodsFromServer],
+    displayList: false,
+    arrayList: [...goodsFromServer],
+  };
+
+  reverseList = () => {
+    this.setState(state => ({
+      arrayList: [...state.arrayList].reverse(),
+    }));
+  };
+
+  sortList = () => {
+    this.setState(state => ({
+      arrayList: [...state.arrayList].sort(),
+    }));
+  };
+
+  sortByLengthList = () => {
+    this.setState(state => ({
+      arrayList: [...state.arrayList].sort((a, b) => a.length - b.length),
+    }));
+  };
+
+  render() {
+    const { goods, displayList, arrayList } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Goods</h1>
+        {displayList === false && (
+        <button
+          type="button"
+          onClick={() => {
+            this.setState({
+              displayList: true,
+            });
+          }}
+        >
+          Start
+        </button>
+        )}
+
+        {displayList === true && (
+        <>
+          <button type="button" onClick={this.reverseList}>Reverse</button>
+
+          <button
+            type="button"
+            onClick={this.sortList}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            onClick={this.sortByLengthList}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            onClick={() => {
+              this.setState({
+                arrayList: [...goods],
```

Comment:
```suggestion
                arrayList: [...goodsFromServer],
```
No need to create identical variable `goods`